### DISCUSSION TITLE
From name not extracted

### DISCUSSION THREAD

**Discussion by FliegenKLATSCH** [rspamd/rspamd#4443] [Q&A]
Rspamd daemon version 3.5 on debian.

Example message:
```
Return-Path: <arwukdm@hostel.armenia.su>
Message-ID: <7432706538138063456015558713061037224525@hostel.armenia.su>
From: "Exodermin" <arwukdm@hostel.armenia.su>
To: <m.jaeschke@klocke-antrieb.de>
Subject: Ein allumfassender Ansatz zur Behandlung von Fusspilz
Date: Mon, 20 Mar 2023 21:39:19 +0200
MIME-Version: 1.0
```

The extracted email:name is empty:
```
2023-03-20 20:53:42 #3767988(controller) <6529fd>; multimap; multimap.lua:631: apply filter email:name for rule BLACKLIST_SENDER_NAME: arwukdm@hostel.armenia.su -> {[1] = }
2023-03-20 20:53:42 #3767988(controller) <6529fd>; multimap; multimap.lua:455: check value  for multimap BLACKLIST_SENDER_NAME
```

> from | matches envelope from (or header From if envelope from is absent)

According to the docs it should take the name from the `From` header? Do I oversee something or is this a bug? 
Also, if I test the selector `from` it always gives me only the email, not the name.. is that expected?
How can I match the name in the `From` header?

**Comment by ghost** [comment]
Seeing the same issue in my rspamd installation.
Have you found some solution?

**Comment by fatalbanana** [comment]
```hcl
BLACKLIST_FROM_DISPLAYNAME {
  # To work with MIME From use `header` type
  type = "header";
  header = "from";
  filter = "email:name";
  map = "file:///tmp/example.map";
  score = 10.0;
}
```

https://rspamd.com/doc/modules/multimap.html#examples

**Reply by ghost** [reply]
Thats exactly the configuration I used, but it does not work.
I suspect to have an issue in my map file, I just inserted the Displaynames like:

First Name Last Name

**Reply by fatalbanana** [reply]
Keys containing spaces [should be quoted](https://rspamd.com/doc/faq.html#what-can-be-in-the-maps)
